app/web/src/components/ChangesPanelHistory.vue

```diff
@@ -1,69 +1,143 @@
-    <ActionsList
-      v-for="[detail, actions] in actionsStore.historyActionsGrouped"
-      :key="detail.changeSetId"
-      :actions="actions"
-      :changeSet="getChangeSet(detail)"
-      :clickAction="clickAction"
-      :selectedFuncRunIds="selectedFuncRunId ? [selectedFuncRunId] : []"
-      kind="history"
-      @history="openHistory"
-    />
+    <TreeNode
+      enableDefaultHoverClasses
+      enableGroupToggle
+      alwaysShowArrow
+      indentationSize="none"
+      leftBorderSize="none"
+      defaultOpen
+      internalScrolling
+      class="min-h-[32px]"
+      primaryIconClasses=""
+      label="Actions"
+    >
+      <!-- TODO(Wendy)- SEARCH BAR SHOULD GO HERE -->
+      <div ref="actionDivRef" class="py-2xs w-full h-full">
+        <template v-if="actionsStore.historyActions.length > 0">
+          <ActionsList
+            v-for="[detail, actions] in actionsStore.historyActionsGrouped"
+            :key="detail.changeSetId"
+            :actions="actions"
+            :changeSet="getChangeSet(detail)"
+            :clickAction="clickActionOrMgmtRun"
+            :selectedFuncRunIds="selectedFuncRunId ? [selectedFuncRunId] : []"
+            kind="history"
+            @history="openHistory"
+          />
+        </template>
+        <EmptyStateCard
+          v-else
+          iconName="actions"
+          primaryText="No Actions Have Been Executed"
+          secondaryText="There is no action history to display for this change set."
+        />
+      </div>
+    </TreeNode>
+    <TreeNode
+      enableDefaultHoverClasses
+      enableGroupToggle
+      alwaysShowArrow
+      indentationSize="none"
+      leftBorderSize="none"
+      defaultOpen
+      internalScrolling
+      class="min-h-[32px]"
+      primaryIconClasses=""
+      label="Management Functions"
+    >
+      <div ref="managementDivRef" class="w-full h-full">
+        <ManagementHistoryList
+          v-if="managementHistoryForChangeSet.length > 0"
+          :managementHistory="managementHistoryForChangeSet"
+          :clickItem="clickActionOrMgmtRun"
+          :funcRunId="selectedFuncRunId"
+          @history="openHistory"
+        />
+        <EmptyStateCard
+          v-else
+          iconName="actions"
+          primaryText="No Management Functions Have Been Executed"
+          secondaryText="There is no management function history to display for this change set."
+        />
+      </div>
+    </TreeNode>
+
     <FuncRunTabGroup
-      :close="deselectAction"
+      :close="deselectActionOrMgmtRun"
       :funcRun="funcRun"
-      :selectedAction="selectedAction"
+      :open="!!selectedFuncRunId"
       :selectedTab="selectedTab"
     />
-  </ScrollArea>
-  <EmptyStateCard
-    v-else
-    iconName="actions"
-    primaryText="No Actions Have Been Taken"
-    secondaryText="There is no action history to display for this change set."
-  />
+  </div>
+  <div v-else ref="actionDivRef" class="py-2xs w-full h-full">
```

Comment:
@zacharyhamm Look here!

Reply:
ahh! i see! very clever

Reply:
that was hard to parse in the diff but it's clear from the full file.